nservicebus/wormhole/index.md

```diff
@@ -0,0 +1,95 @@
+| Routing                 | set destination site when sending       | <ul><li>set destination site via routing config</li><li>set destination site from message content</li></ul> |
+| Failure handling        | <ul><li>immediate retries</li><li>delayed retries</li></ul> | <ul><li>immediate retries</li></ul>                                 |
+| Automatic deduplication | built-in via persistence                | no                                                                                      |
+|                         |                                         |                                                                                         |
+|                         |                                         |                                                                                         |
+|                         |                                         |                                                                                         |
+
+## Topology
+
+The Wormhole-enabled system consists of *sites*. Sites can be geographically distributed. It is assumed that the connectivity within the site is much better than between the sites. Each site forms an independent NServiceBus system with its own set of infrastructure e.g. ServiceControl and ServicePulse.
+
+Each site hosts a Wormhole *gateway* which is a special node that allows to send messages to remote sites. Any given message type can be configured to be routed to the gateway (instead of to a regular endpoint). 
+
+Wormhole *tunnels* connect the gateways. Tunnels are always single-hop so a message always travels through exactly two gateways: it enters the tunnel through a gateway in the origin site and leaves it through a gateway in in the destination site.
+
+The transport used to implement the tunnels need to be the same (and be configured with the same parameters) for all the gateways that take part in a given system.
+
+
+## Transports
+
+The key advantage of Wormhole is the ability to run any NServiceBus transport, in addition to built-in HTTP transport, for exchanging messages between the sites. For interoperability reasons the most useful transports for the Wormhole are these based on HTTP protocol.
+
+
+### HTTP
+
+The HTTP transport runs a HTTP listener in each Wormhole gateway in a similar way to the standard NServiceBus Gateway. The advantage of listener-based approach is immunity to connection problems in the tunnel. When the connection is down the messages queue up on the sender side, waiting to be forwarded.
+
+The disadvantage is the need to expose a HTTP endpoint in the public network.
+
+
+### ASB and ASQ
+
+As an alternative to the HTTP transport, Azure transports (ASB or ASQ) can be used as a tunnel transport. The advantage of Azure transports is ability to run the Wormhole nodes behind firewall without the need to open any ports as the ASB/ASQ client does not run the HTTP listener.
```

Comment:
Don't use the ASB/ASQ acronyms

I'd recommend "Azure Transports" for the header so it's not super long

> run the Wormhole nodes behind firewall

run the Wormhole nodes behind **a** firewall